src/ui/settings/__tests__/index.js

```diff
@@ -0,0 +1,345 @@
+import sinon from 'sinon';
+import expect from 'expect.js';
+import init from '..';
+import defaultsProvider from '../defaults';
+
+describe('ui settings', function () {
+  describe('overview', function () {
+    it('has expected api surface', function () {
+      const { uiSettings } = instantiate();
+      expect(typeof uiSettings.get).to.be('function');
+      expect(typeof uiSettings.getAll).to.be('function');
+      expect(typeof uiSettings.getDefaults).to.be('function');
+      expect(typeof uiSettings.getRaw).to.be('function');
+      expect(typeof uiSettings.getUserProvided).to.be('function');
+      expect(typeof uiSettings.remove).to.be('function');
+      expect(typeof uiSettings.set).to.be('function');
+      expect(typeof uiSettings.setMany).to.be('function');
+    });
+  });
+
+  describe('#setMany()', function () {
+    it('updates a single value in one operation', function () {
+      const { server, uiSettings, configGet } = instantiate();
+      const result = uiSettings.setMany({ one: 'value' });
+      expect(typeof result.then).to.be('function');
```

Comment:
I think these tests can be simplified by creating a separate "returns a promise" test that makes this kind of assertion instead of repeating it within each test. This will make each test's description more accurate, make the tests more readable since they will be testing "just one thing", and the tests themselves will be deflated a bit.

```javascript
describe('#setMany()', () => {
    it('returns a promise', () => {
      const { uiSettings } = instantiate();
      const result = uiSettings.setMany({});
      expect(typeof result.then).to.be('function');
    });

    it('updates a single value in one operation', () => {
      /* ... */
    });

    it('updates several values in one operation', () => {
      /* ... */
    });
  });
```